.github/workflows/verify.yaml

```diff
@@ -2,12 +2,26 @@ name: Test Code (Style, Tests)
 
 on:
   push:
+    branches:
+      - main
   pull_request:
     paths-ignore:
       - '**.md'
       - 'docs/**'
 
 jobs:
+  Checkstyle:
+    permissions:
+      id-token: write
+      checks: write
+      pull-requests: write
```

Comment:
why are these permissions needed?

Reply:
It's the same set of permissions as granted in EDC. I assume "pull-request" is required to show Checkstyle errors in the PR? But I am not sure... I will try to remove them and see what happens.

Reply:
Seems more like a leftover from using a PR action previously

Reply:
Note this https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/1651

Reply:
checks and pull-request write permission is needed by unit test result publishing action as per the official doc https://github.com/EnricoMi/publish-unit-test-result-action#permissions.

So, this permission can be moved where we have this action(EnricoMi/publish-unit-test-result-action).

Reply:
I have added these permissions to the "Test" job for correctness, although it has been working without them anyways.